src/main/java/hw17/MainApp.java

```diff
@@ -0,0 +1,91 @@
+package hw17;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+import static hw17.ProductType.*;
+
+public class MainApp {
+    public static void main(String[] args) {
+        List<Product> products = addProductsList();
+        products.forEach(System.out::println);
+
+        getBookPrice250(products).forEach(System.out::println);
+
+        getBookDiscount(products).forEach(System.out::println);
+
+        getBookCheapest(products).forEach(System.out::println);
+
+        getThreeMostRecentlyAdded(products).forEach(System.out::println);
+
+        System.out.println("\nThe total cost of BOOK (price >75) for this year: " +
+                getCalSomeProductsThisYear(products));
+    }
+
+    private static List<Product> getBookPrice250(List<Product> products) {
+        System.out.println("\nAll BOOKS with price > 250:");
+        return products.stream()
+                .filter(product -> BOOK == product.getProductType())
+                .filter(product -> product.getPrice() > 250)
+                .toList();
+    }
+
+    private static List<Product> getBookDiscount(List<Product> products) {
+        System.out.println("\nAll BOOKS with discount (10%):");
+        return products.stream()
+                .filter(product -> BOOK == product.getProductType())
+                .filter(Product::isDiscount)
+                .peek(product -> product.setPrice(product.getPrice() * 0.9))
+                .toList();
+    }
+
+    private static List<Product> getBookCheapest(List<Product> products) {
+        System.out.println("\nCheapest product:");
+        try {
+            return products.stream()
+                    .filter(product -> BOOK == product.getProductType())
+                    .min(Comparator.comparing(Product::getPrice))
+                    .stream().toList();
+        } catch (Exception e) {
+            throw new NullPointerException(e.getMessage() + "\nProduct BOOK not found");
+        }
+    }
+
+    private static List<Product> getThreeMostRecentlyAdded(List<Product> products) {
+        System.out.println("\nThree latest added products:");
+        return products.stream()
+                .sorted(Comparator.comparing(Product::getDateAdded).reversed())
+                .limit(3)
+                .toList();
+    }
+
+    private static double getCalSomeProductsThisYear(List<Product> products) {
+        return products.stream()
+                .filter(product -> product.dateAdded.compareTo(LocalDate.of(2023, 1, 1)) >= 0)
+                .filter(product -> BOOK.equals(product.getProductType()))
```

Comment:
In one place you use ==, in other `equals`. Better to implement all in one style